test/components/legendTests.ts

```diff
@@ -355,4 +355,20 @@ describe("Legend", () => {
 
     svg.remove();
   });
+
+  it("symbol() passes index correctly", () => {
+    var domain = ["AA", "BB", "CC"];
+    color.domain(domain);
+
+    var expectedIndex = 0;
+    var symbolChecker = (d: any, index: number) => {
+      assert.strictEqual(index, expectedIndex, "message");
```

Comment:
When I originally typed `"message"`, I originally meant that as an example ... I should have made myself clearer.